### PR TITLE
use interop for getReceiver

### PR DESCRIPTION
**Pull request by Haiyang-Sun**
Using the interop library to look for the recever of the dynamic scope.
Refer to [ScopeInstrumentTest](https://github.com/graalvm/graaljs/blob/master/graal-js/src/com.oracle.truffle.js.test.instrumentation/src/com/oracle/truffle/js/test/instrumentation/ScopeInstrumentTest.java)

### CODE REVIEW

File: src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/FunctionRootEventHandler.java [hunk 1]
```diff
@@ -47,9 +44,8 @@
 
     protected final String builtinName;
 
-    @CompilationFinal private FrameSlot thisSlot;
-
-    @CompilationFinal private boolean thisSlotInitialized = false;
+    private static final NodeLibrary NODE_LIBRARY = NodeLibrary.getUncached();
+    private static final InteropLibrary INTEROP = InteropLibrary.getUncached();
```
Comment (eleinadani) [inline src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/FunctionRootEventHandler.java:48]: using uncached interop libraries is quite bad for performance: couldn't you store them in a node and use the cached version instead? (see e.g. https://github.com/oracle/graal/blob/master/truffle/docs/InteropMigration.md)

Comment (alexjordan) [reply src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/FunctionRootEventHandler.java:59]: I added the caching in 531c7e9 and also brought back the old optimization that reads `<this>` from the frame slots (which does trigger in our tests). This seemed to shave 2-3 seconds off the `lodash` test when I tested it. Unfortunately that's not a proper benchmark, maybe running the V8 tests with the compiler would be a better option.

Comment (alexjordan) [reply src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/FunctionRootEventHandler.java:59]: @eleinadani does that caching look okay?

**Review by alexjordan — APPROVED** [review]
@Haiyang-Sun anything that needs changing or good to merge?

File: src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/FunctionRootEventHandler.java (hunk 1 above)
Comment (eleinadani) [reply src/ch.usi.inf.nodeprof/src/ch/usi/inf/nodeprof/handlers/FunctionRootEventHandler.java:59]: uh, I see... let's keep it like that then. I think I should do a full review of the entire codebase and look into how interop is used anyway at some point

**Pull request merged**